static/app/components/events/interfaces/performance/spanEvidenceKeyValueList.tsx

```diff
@@ -1,97 +1,125 @@
+import kebabCase from 'lodash/kebabCase';
+
 import {t} from 'sentry/locale';
-import {IssueType, KeyValueListData} from 'sentry/types';
+import {IssueType, KeyValueListData, KeyValueListDataItem} from 'sentry/types';
 
 import KeyValueList from '../keyValueList';
 import {RawSpanType} from '../spans/types';
 
 import {TraceContextSpanProxy} from './spanEvidence';
 
+type Span = RawSpanType | TraceContextSpanProxy;
+
 type SpanEvidenceKeyValueListProps = {
   issueType: IssueType | undefined;
-  offendingSpans: Array<RawSpanType | TraceContextSpanProxy>;
-  parentSpan: RawSpanType | TraceContextSpanProxy | null;
+  offendingSpans: Span[];
+  parentSpan: Span | null;
   transactionName: string;
 };
 
 const TEST_ID_NAMESPACE = 'span-evidence-key-value-list';
 
-export function SpanEvidenceKeyValueList({
-  issueType,
-  offendingSpans,
-  parentSpan,
-  transactionName,
-}: SpanEvidenceKeyValueListProps) {
-  const data: KeyValueListData = [
+export function SpanEvidenceKeyValueList(props: SpanEvidenceKeyValueListProps) {
+  if (!props.issueType) {
+    return <DefaultSpanEvidence {...props} />;
+  }
+
+  const Component =
     {
-      key: '0',
-      subject: t('Transaction'),
-      value: transactionName,
-      subjectDataTestId: `${TEST_ID_NAMESPACE}.transaction-name`,
-    },
-  ];
-
-  if (issueType === IssueType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES) {
-    data.push({
-      key: '1',
-      subject: t('Parent Span'),
-      value: getSpanEvidenceValue(parentSpan),
-      subjectDataTestId: `${TEST_ID_NAMESPACE}.parent-name`,
-    });
-
-    data.push({
-      key: '2',
-      subject: t('Repeating Span'),
-      value: getSpanEvidenceValue(offendingSpans[0]),
-      subjectDataTestId: `${TEST_ID_NAMESPACE}.offending-spans`,
-    });
-  } else if (issueType === IssueType.PERFORMANCE_N_PLUS_ONE_API_CALLS) {
-    data.push({
-      key: '1',
-      subject: t('Offending Span'),
-      value: getSpanEvidenceValue(offendingSpans[0]),
-      subjectDataTestId: `${TEST_ID_NAMESPACE}.offending-spans`,
-    });
-
-    let problemParameters;
-    if (issueType === IssueType.PERFORMANCE_N_PLUS_ONE_API_CALLS) {
-      problemParameters = getProblemParameters(offendingSpans);
-    }
+      [IssueType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES]: NPlusOneDBQueriesSpanEvidence,
+      [IssueType.PERFORMANCE_N_PLUS_ONE_API_CALLS]: NPlusOneAPICallsSpanEvidence,
+      [IssueType.PERFORMANCE_SLOW_SPAN]: SlowSpanSpanEvidence,
+    }[props.issueType] ?? DefaultSpanEvidence;
```

Comment:
I'm not too familiar with this syntax at the end here, is this essentially the same as using a ternary operator to set `Component` to `DefaultSpanEvidence` if `props.issueType` is nullish?

Reply:
Yup, exactly, but `undefined` rather than `null`